iroh-net/src/magicsock.rs

```diff
@@ -2090,26 +2046,30 @@ impl Actor {
         // The STUN address(es) are always first.
         // Despite this sorting, clients are not relying on this sorting for decisions;
 
-        if self.inner.endpoints.lock().set(&eps) {
+        if self.inner.endpoints.write().set(&eps) {
             log_endpoint_change(&eps);
             if let Some(ref cb) = self.inner.on_endpoints {
                 cb(&eps[..]);
             }
+            for (public_key, region_id) in self.inner.pending_call_me_maybes.lock().drain() {
+                self.inner
+                    .send_or_queue_call_me_maybe(region_id, public_key);
```

Comment:
shouldn't this force sending instead of potentially requeuing? not sure in either direction

Reply:
Yea it won't requeue because endpoints.set was just called which means endpoints.is_fresh will always be true.
We could add a separate send_call_me_maybe method for clarity though.

Reply:
I think that would be good, to make the intent clear (given the complexity of this code.. )

Reply:
Changed it to separate functions.